src/NUnitEngine/nunit.engine.netstandard/Runners/DirectTestRunner.cs

```diff
@@ -0,0 +1,162 @@
+﻿// ***********************************************************************
+// Copyright (c) 2011-2014 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using NUnit.Engine.Extensibility;
+using NUnit.Engine.Internal;
+using NUnit.Engine.Drivers;
+using NUnit.Engine.Services;
+
+namespace NUnit.Engine.Runners
+{
+    /// <summary>
+    /// DirectTestRunner is the abstract base for runners 
+    /// that deal directly with a framework driver.
+    /// </summary>
+    public class DirectTestRunner : AbstractTestRunner
```

Comment:
Any engine files that were dramatically different I created a new copy for the .NET Standard engine. They were getting way too complicated with the `#if` statements.